### ISSUE TITLE
bug(config-api): Blacklist some urls from webhook for security reason

### ISSUE THREAD

**Issue by syntrydy** [JanssenProject/jans#8574]
-------------------

### Description

Currently there is no security in webhook feature. 
The following should be check:

1. Perform URL validation(url should be valid url and resolvable).
1. Ensure URL starts with "https://", disallow "file://" and other non-HTTPS schemes.
1. Block typical local IPs: 127.0.x, 192.168.x, 172.x.
1. Prohibit "localhost" and "http://"
1. Require a specific response header for POST requests, unique to the customer.

**Comment by 433B** [comment]
Hello! I'm interested in this topic, could I have it for myself? I have experience in this area.
Additionally, this is a key assignment for one of my university courses.

**Comment by nynymike** [comment]
Sure... why not. I don't think it's a super high priority at Jans.

**Comment by syntrydy** [comment]
@nynymike 

Before the source app sends an HTTP request via the webhook, it hashes the payload (request body) with HMAC using the secret key. The resulting hash is then bundled into the HTTP request as a header, and the entire request (header and body) is sent to the webhook endpoint.

Upon receiving the HTTP request, the destination app hashes the body with the secret key and then compares the result to the hash provided in the header. If the values match, the destination app knows the data is legit and processes it. If the values do not match, the destination app rejects the data and executes whatever code was written for that scenario — perhaps creating a log entry or sending a notification.

**Comment by 433B** [comment]
Hi, @nynymike @syntrydy, I understand correctly that the logic you described earlier is in this place?
```java
RegisterValidator.validateRequestObject();
```

**Comment by 433B** [comment]
Hello @syntrydy @nynymike @moabu, 
I am still analyzing your project as it is very large. I have already found a place where I want to implement this logic. The url blocking will be at the configuration level in: _### config-jans-auth.properties_. so that it can be easily changed in the future. In this place - ### _RegisterValidator.validateRequestObjectHmac()_, I would like to implement this logic to block url, as described by a colleague above. Could you please tell me if I can use BaseClient to take the url?

**Comment by syntrydy** [comment]
@433B any update?

**Comment by 433B** [comment]
> [@433B](https://github.com/433B) any update?

@syntrydy Hi, yes, I just need to finalize the mock tests for my implementation and it's done. I wanted to commit my code yesterday, but got an access error with 403 status. Do you know why this happens?

**Comment by syntrydy** [comment]
@433B 403 is authorization issue. You can check with @nynymike  or just open a pull request.

**Comment by 433B** [comment]
Hi, @nynymike I still get the access error, could you provide me with them?

**Comment by 433B** [comment]
@nynymike Could you please provide me with access to your project?
I can't do commit

**Comment by ossdhaval** [comment]
@syntrydy Sorry for joining this conversation late. Do we have any documentation around webhooks? If so can you share a link to it here? Let's mention these changes in the docs.

**Comment by ossdhaval** [comment]
@syntrydy I was thinking of this proposal from `developer setup` and `local testing` perspective. 

In both scenarios above, the use of `localhost` and IPs like `127.`  often come handy. See if this concern is relevant to this feature.

**Comment by yuriyz** [comment]
Thanks @ossdhaval for asking clarification. I see PR opened agains jans-auth-server which for some reason throws misleading exception if signature verification failed. https://github.com/JanssenProject/jans/pull/10100

@syntrydy is ticket about webhooks in config-api or about something else? (https://github.com/JanssenProject/jans/blob/09bbd66f9159c6458d14b20876c277151a1f8452/jans-config-api/plugins/admin-ui-plugin/src/main/java/io/jans/ca/plugin/adminui/rest/webhook/WebhookResource.java#L43)

**Comment by 433B** [comment]
Thank you, @yuriyz , for your review. It seems I may have misunderstood the task. I implemented the logic as described by @syntrydy above:
"If the values don't match, the destination app rejects the data and executes the appropriate code for that scenario, such as creating a log entry or sending a notification."

**Comment by yuriyz** [comment]
@433B Right, it looks like misunderstanding. Please close that PR and lets get requirements listed in this ticket description for clarity.

**Comment by 433B** [comment]
@yuriyz could you check please my new pr?

**Comment by yuriyz** [comment]
@433B sure, I've commented to PR, please check.
